multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/steps/ValidateDeployParametersStep.java

```diff
@@ -57,22 +59,34 @@ protected String getStepErrorMessage(ProcessContext context) {
     }
 
     private void validateParameters(ProcessContext context) {
-        validateExtensionDescriptorFileIds(context);
-        validateFilesSizeLimit(context);
-        validateArchive(context);
+        List<FileEntry> extensionDescriptors = validateExtensionDescriptorFileIds(context);
+        List<FileEntry> archivePartEntries = getArchivePartEntries(context);
+
+        if (archivePartEntries.size() > 1) {
+            validateFilesSizeLimit(context, archivePartEntries, extensionDescriptors);
+            mergeArchive(context, archivePartEntries);
+        } else {
+            getStepLogger().infoWithoutProgressMessage(Messages.ARCHIVE_WAS_NOT_SPLIT_TOTAL_SIZE_IN_BYTES_0, archivePartEntries.get(0)
+                                                                                                                               .getSize());
```

Comment:
if there is only one archive part, the size of the ext descriptors + archive will not be validated